src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -251,8 +251,26 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             "ptr_offset_from" => {
-                let a = self.read_immediate(args[0])?.to_scalar()?.to_ptr()?;
-                let b = self.read_immediate(args[1])?.to_scalar()?.to_ptr()?;
+                let isize_layout = self.layout_of(self.tcx.types.isize)?;
+                let a = self.read_immediate(args[0])?.to_scalar()?;
+                let b = self.read_immediate(args[1])?.to_scalar()?;
+
+                // Special case: if both scalars are *equal integers*
+                // and not NULL, their offset is 0.
```

Comment:
why is the null case important? The docs state nothing of the kind. Or is the "valid object" thing the problem?

Reply:
Actually, since `wrapping_offset_from` is safe, and you can supply null pointers to it... I guess this should be allowed?

Reply:
Indeed this is about the "valid object" part. There's never a valid object at 0, not even one of size 0, so 0 is never a valid input here.

`wrapping_offset_from` does not have the "valid object" requirement.